packages/hdwallet-phantom/src/phantom.ts

```diff
@@ -0,0 +1,447 @@
+    return false;
+  }
+
+  public supportsBroadcast(): boolean {
+    return true;
+  }
+
+  public describePath(msg: core.DescribePath): core.PathDescription {
+    switch (msg.coin) {
+      case "Ethereum":
+        return eth.describeETHPath(msg.path);
+      default:
+        throw new Error("Unsupported path");
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public ethNextAccountPath(msg: core.ETHAccountPath): core.ETHAccountPath | undefined {
+    // TODO: What do we do here?
+    return undefined;
+  }
+
+  public async ethSupportsNetwork(chainId: number): Promise<boolean> {
+    return chainId === 1;
+  }
+
+  public async ethSupportsSecureTransfer(): Promise<boolean> {
+    return false;
+  }
+
+  public ethSupportsNativeShapeShift(): boolean {
+    return false;
+  }
+
+  public async ethSupportsEIP1559(): Promise<boolean> {
+    return true;
+  }
+
+  public ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+    return eth.ethGetAccountPaths(msg);
+  }
+}
+
+export class PhantomHDWallet implements core.HDWallet, core.ETHWallet {
+  readonly _supportsETH = true;
+  readonly _supportsETHInfo = true;
+  readonly _supportsBTCInfo = false;
+  readonly _supportsBTC = true;
+  readonly _supportsCosmosInfo = false;
+  readonly _supportsCosmos = false;
+  readonly _supportsEthSwitchChain = false;
+  readonly _supportsAvalanche = false;
+  readonly _supportsOptimism = false;
+  readonly _supportsBSC = false;
+  readonly _supportsPolygon = true;
+  readonly _supportsGnosis = false;
+  readonly _supportsArbitrum = false;
+  readonly _supportsArbitrumNova = false;
+  readonly _supportsBase = false;
+  readonly _supportsOsmosisInfo = false;
+  readonly _supportsOsmosis = false;
+  readonly _supportsBinanceInfo = false;
+  readonly _supportsBinance = false;
+  readonly _supportsDebugLink = false;
+  readonly _isPortis = false;
+  readonly _isPhantom = true;
+  readonly _supportsRippleInfo = false;
+  readonly _supportsRipple = false;
+  readonly _supportsEosInfo = false;
+  readonly _supportsEos = false;
+  readonly _supportsFioInfo = false;
+  readonly _supportsFio = false;
+  readonly _supportsThorchainInfo = false;
+  readonly _supportsThorchain = false;
+  readonly _supportsSecretInfo = false;
+  readonly _supportsSecret = false;
+  readonly _supportsKava = false;
+  readonly _supportsKavaInfo = false;
+  readonly _supportsTerra = false;
+  readonly _supportsTerraInfo = false;
+
+  info: PhantomHDWalletInfo & core.HDWalletInfo;
+  ethAddress?: string | null;
+  evmProvider: any;
+  bitcoinProvider: any;
+
+  constructor(evmProvider: unknown, bitcoinProvider: unknown) {
+    this.info = new PhantomHDWalletInfo();
+    this.evmProvider = evmProvider;
+    this.bitcoinProvider = bitcoinProvider;
+  }
+
+  async getFeatures(): Promise<Record<string, any>> {
+    return {};
+  }
+
+  public async isLocked(): Promise<boolean> {
+    return !this.evmProvider._phantom.isUnlocked();
```

Comment:
This won't work, it should live under `_metamask`:

https://github.com/phantom/multi-chain-sandbox/blob/be63de7ee03496ed71c44fc886fa49b23f04c98e/src/types.ts#L61-L63

<img width="258" alt="image" src="https://github.com/user-attachments/assets/6244932a-2066-473d-87df-f013e2052505">